# Atividade de vendas no Brasil

##### Arthur M. C. Pereira

##### Importando bibliotecas e lendo o CSV


In [197]:
import pandas as pd 
import seaborn as sns 
import numpy as np
import matplotlib as plt

In [29]:
df=pd.read_csv('../../data/vendas_brasil_1.csv')
pd.set_option('display.max_columns', None)
df.head()

,data_venda,mes,cidade,canal_venda,categoria,produto,vendedor,fidelidade_cliente,preco_unitario,custo_unitario,margem_pct,quantidade,desconto_pct,receita_total,frete,avaliacao_cliente,prazo_entrega_dias,devolucao,entrega_no_prazo,alta_temporada,flag_outlier_preco
0,2022-04-13,4,Rio de Janeiro,App Mobile,Eletrônicos,Notebook,Bruno Souza,Diamante,739.39,561.75,22.31,10,5.2,6933.48,8.68,4.92,5.0,False,True,False,False
1,2023-03-12,3,Manaus,Site Web,Vestuário,Camiseta,Elisa Costa,Bronze,187.48,135.15,30.26,5,0.6,930.28,NaN,4.98,19.0,True,False,False,False
2,2022-09-28,9,Goiânia,App Mobile,Vestuário,Tênis,João Vieira,Bronze,76.11,61.40,21.61,14,0.7,1058.50,7.98,3.47,NaN,True,True,False,False
3,2022-04-17,4,Fortaleza,Site Web,Vestuário,Calça Jeans,Carla Mendes,Bronze,145.86,88.53,41.27,11,0.6,1592.43,20.49,4.50,16.0,False,True,False,False
4,2022-03-13,3,Belo Horizonte,App Mobile,Esportes,Bicicleta,João Vieira,Bronze,133.92,107.25,9.47,1,19.5,107.64,22.89,3.49,13.0,False,True,False,False


In [30]:
df.columns

Index(['data_venda', 'mes', 'cidade', 'canal_venda', 'categoria', 'produto',
       'vendedor', 'fidelidade_cliente', 'preco_unitario', 'custo_unitario',
       'margem_pct', 'quantidade', 'desconto_pct', 'receita_total', 'frete',
       'avaliacao_cliente', 'prazo_entrega_dias', 'devolucao',
       'entrega_no_prazo', 'alta_temporada', 'flag_outlier_preco'],
      dtype='str')

## Entregável 1 NumPy

##### A. Criando arrays

In [31]:
preco_unit = np.array(df['preco_unitario'])
custo_unit = np.array(df['custo_unitario'])
margem_pct = np.array(df['margem_pct'])
quantidade = np.array(df['quantidade'])
desconto_pct = np.array(df['desconto_pct'])
receita_total = np.array(df['receita_total'])
frete = np.array(df['frete'])
avaliacao_cliente= np.array(df['avaliacao_cliente'])


##### B. Cálculos com operações vetorizadas

In [32]:
'''1. valor bruto da venda
2. valor do desconto
3. custo total
4. lucro estimado'''

valor_bruto_venda = preco_unit * quantidade
valor_desconto = valor_bruto_venda * desconto_pct/100
custo_total = custo_unit * quantidade
lucro_estimado = receita_total - custo_total 
markup = preco_unit / custo_unit

valor_bruto_venda, valor_desconto, custo_total, lucro_estimado, markup

(array([7393.9 ,  937.4 , 1065.54, ...,  386.05, 1379.63, 4229.64],
       shape=(37617,)),
 array([384.4828 ,   5.6244 ,   7.45878, ...,   0.     , 150.37967,
        422.964  ], shape=(37617,)),
 array([5617.5 ,  675.75,  859.6 , ...,  258.58, 1170.61, 3482.76],
       shape=(37617,)),
 array([1315.98,  254.53,  198.9 , ...,  125.86,  789.57,  312.97],
       shape=(37617,)),
 array([1.31622608, 1.38719941, 1.23957655, ..., 1.49296156, 1.17855648,
        1.21445061], shape=(37617,)))

##### C. Máscaras booleanas

In [33]:
np.nanmean(receita_total)
vendas_acima_media = receita_total > np.nanmean(receita_total)
prejuizo = lucro_estimado < 0 
desconto_alto = desconto_pct > 0.3


In [34]:
vendas_acima_media = receita_total > np.nanmean(receita_total)
print(vendas_acima_media)
print(receita_total[vendas_acima_media])
print(np.nanmean(receita_total[vendas_acima_media]))

[ True False False ... False  True  True]
[6933.48 2167.48 7945.96 ... 3369.62 1960.18 3795.73]
5688.244210697271


In [35]:
print(np.nanmean(lucro_estimado))
print(lucro_estimado[prejuizo])
print(np.nanmean(lucro_estimado[prejuizo]))

513.187152883005
[ -14.01 -159.89  -42.12 ...  -27.31 -264.88  -29.32]
-73.85957103321033


##### D. Agregações

In [45]:
print(f'a média da receita foi: {round(np.nanmean(receita_total), 2)}')
print(f'a maior receita foi: {receita_total.max()}')
print(f'a menor receita foi: {receita_total.min()}')
print(f'o lucro médio foi: {round(np.nanmean(lucro_estimado), 2)}')
print(f'o maior lucro foi: {round(lucro_estimado.max(), 2)}')
print(f'a média da quantidade de produtos vendidos foi: {round(np.nanmean(quantidade), 2)}')

a média da receita foi: 1891.69
a maior receita foi: 58430.28
a menor receita foi: 3.7
o lucro médio foi: 513.19
o maior lucro foi: 19986.7
a média da quantidade de produtos vendidos foi: 7.48


## Entregável 2 Panda

##### A. Primeiro olhar no dataset

In [ ]:
df.head()

,data_venda,mes,cidade,canal_venda,categoria,produto,vendedor,fidelidade_cliente,preco_unitario,custo_unitario,margem_pct,quantidade,desconto_pct,receita_total,frete,avaliacao_cliente,prazo_entrega_dias,devolucao,entrega_no_prazo,alta_temporada,flag_outlier_preco
0,2022-04-13,4,Rio de Janeiro,App Mobile,Eletrônicos,Notebook,Bruno Souza,Diamante,739.39,561.75,22.31,10,5.2,6933.48,8.68,4.92,5.0,False,True,False,False
1,2023-03-12,3,Manaus,Site Web,Vestuário,Camiseta,Elisa Costa,Bronze,187.48,135.15,30.26,5,0.6,930.28,NaN,4.98,19.0,True,False,False,False
2,2022-09-28,9,Goiânia,App Mobile,Vestuário,Tênis,João Vieira,Bronze,76.11,61.40,21.61,14,0.7,1058.50,7.98,3.47,NaN,True,True,False,False
3,2022-04-17,4,Fortaleza,Site Web,Vestuário,Calça Jeans,Carla Mendes,Bronze,145.86,88.53,41.27,11,0.6,1592.43,20.49,4.50,16.0,False,True,False,False
4,2022-03-13,3,Belo Horizonte,App Mobile,Esportes,Bicicleta,João Vieira,Bronze,133.92,107.25,9.47,1,19.5,107.64,22.89,3.49,13.0,False,True,False,False


In [ ]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 37617 entries, 0 to 37616
Data columns (total 21 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   data_venda          37617 non-null  str    
 1   mes                 37617 non-null  int64  
 2   cidade              37617 non-null  str    
 3   canal_venda         37567 non-null  str    
 4   categoria           37617 non-null  str    
 5   produto             37617 non-null  str    
 6   vendedor            37577 non-null  str    
 7   fidelidade_cliente  37617 non-null  str    
 8   preco_unitario      37617 non-null  float64
 9   custo_unitario      37617 non-null  float64
 10  margem_pct          37535 non-null  float64
 11  quantidade          37617 non-null  int64  
 12  desconto_pct        37617 non-null  float64
 13  receita_total       37617 non-null  float64
 14  frete               35738 non-null  float64
 15  avaliacao_cliente   34662 non-null  float64
 16  prazo_entrega_d

In [51]:
df.describe()


,mes,preco_unitario,custo_unitario,margem_pct,quantidade,desconto_pct,receita_total,frete,avaliacao_cliente,prazo_entrega_dias
count,37617.000000,37617.000000,37617.000000,37535.000000,37617.000000,37617.000000,37617.000000,35738.000000,34662.000000,36113.000000
mean,6.517691,253.214933,183.620999,22.714776,7.481750,9.494048,1891.688173,29.885737,3.912094,10.017113
std,3.439296,316.742367,231.376757,8.670213,4.322273,9.097484,2972.445759,24.765161,0.932256,5.477487
min,1.000000,4.940000,3.010000,-0.570000,1.000000,0.000000,3.700000,3.750000,1.000000,1.000000
25%,4.000000,68.960000,49.550000,16.380000,4.000000,0.300000,364.280000,12.190000,3.490000,5.000000
50%,7.000000,111.300000,80.990000,22.700000,7.000000,9.000000,789.570000,22.250000,4.010000,10.000000
75%,9.000000,257.680000,186.570000,29.250000,11.000000,19.000000,1853.210000,39.480000,4.520000,15.000000
max,12.000000,9008.600000,6578.260000,44.270000,161.000000,26.000000,58430.280000,243.820000,5.000000,19.000000


In [53]:
df.shape


(37617, 21)

In [54]:
df.isna().sum()

data_venda               0
mes                      0
cidade                   0
canal_venda             50
categoria                0
produto                  0
vendedor                40
fidelidade_cliente       0
preco_unitario           0
custo_unitario           0
margem_pct              82
quantidade               0
desconto_pct             0
receita_total            0
frete                 1879
avaliacao_cliente     2955
prazo_entrega_dias    1504
devolucao                0
entrega_no_prazo         0
alta_temporada           0
flag_outlier_preco       0
dtype: int64

Perguntas
1. Quantas linhas e colunas existem?
2. Existem dados faltantes?
3. Quais colunas parecem numéricas?
4. Quais colunas parecem categóricas?

Respostas:
1. São 37617 linhas e 21 colunas, conforme mostra o "df.shape".
2. Sim. Há por exemplo 50 dados faltantes na coluna canal_venda, conforme mostra "df.isna().sum()".
3. As colunas "quantidade" e "custo_unitario" parecem ser númericas pelo tipo de dado contido nelas ser int64 e float64.
4. As colunas "produto" e "devolução" parecem ser categóricas pelo tipo de dado contido nelas ser str e bool. 


##### B. Limpeza e tratamento

In [65]:
df['data_venda'] = pd.to_datetime(df['data_venda'],  format="mixed")
df['data_venda']

0       2022-04-13
1       2023-03-12
2       2022-09-28
3       2022-04-17
4       2022-03-13
           ...    
37612   2023-12-31
37613   2023-12-31
37614   2023-12-31
37615   2023-12-31
37616   2023-12-31
Name: data_venda, Length: 37617, dtype: datetime64[us]

In [73]:
df.columns

Index(['data_venda', 'mes', 'cidade', 'canal_venda', 'categoria', 'produto',
       'vendedor', 'fidelidade_cliente', 'preco_unitario', 'custo_unitario',
       'margem_pct', 'quantidade', 'desconto_pct', 'receita_total', 'frete',
       'avaliacao_cliente', 'prazo_entrega_dias', 'devolucao',
       'entrega_no_prazo', 'alta_temporada', 'flag_outlier_preco'],
      dtype='str')

In [ ]:
df['fidelidade_cliente'].unique() #descobrindo qual coluna posso normalizar

<StringArray>
['Diamante',   'Bronze',    'Prata',     'Ouro',     'ouro',    'prata',
   'bronze', 'DIAMANTE',    'PRATA',   'BRONZE',     'OURO', 'diamante']
Length: 12, dtype: str

In [ ]:
df['fidelidade_cliente'] = df['fidelidade_cliente'].str.lower()

In [92]:
df['fidelidade_cliente'].unique()

<StringArray>
['diamante', 'bronze', 'prata', 'ouro']
Length: 4, dtype: str

In [161]:
df['cidade'].unique()

<StringArray>
[ 'Rio de Janeiro',          'Manaus',         'Goiânia',       'Fortaleza',
  'Belo Horizonte',          'Recife',       'São Paulo',    'Porto Alegre',
        'Salvador',        'Curitiba',        'salvador', 'Belo Horizonte ',
  'Rio de janeiro',         'Curtiba',       'Curitiba ',              'SP',
      ' Fortaleza',        'curitiba',              'RJ',        'S. Paulo',
       ' Salvador',         'Manaus ',       'São paulo',      'Fortaleza ',
       'Sao Paulo',              'BH', 'Rio de Janeiro ', ' Belo Horizonte',
         ' Recife',        'Slavador',  'Belo horizonte',             'Rio',
       ' Curitiba',         ' Manaus',        ' Goiânia',         'Recife ',
    'B. Horizonte',   'Porto Alegre ',  'Rio De Janeiro',      'São Paulo ',
        'Goiânia ',   ' Porto Alegre',       'Salvador ', ' Rio de Janeiro',
      ' São Paulo']
Length: 45, dtype: str

In [164]:
df['cidade'] = df['cidade'].str.strip().str.lower()
padraoinator = {
    'rj':'rio de janeiro','rio':'rio de janeiro',
    'sao paulo': 'são paulo', 'sp': 'são paulo', 's. paulo': 'são paulo',
    'bh': 'belo horizonte', 'b. horizonte': 'belo horizonte',
    'slavador': 'salvador',
    'curtiba': 'curitiba'}

df['cidade'] = df['cidade'].replace(padraoinator)
df['cidade'].unique()

<StringArray>
['rio de janeiro',         'manaus',        'goiânia',      'fortaleza',
 'belo horizonte',         'recife',      'são paulo',   'porto alegre',
       'salvador',       'curitiba']
Length: 10, dtype: str

In [111]:
df['vendedor'].value_counts()

vendedor
João Vieira        3817
Ana Lima           3812
Elisa Costa        3811
Bruno Souza        3775
Gabi Alves         3772
Diego Ferreira     3738
Henrique Dias      3734
Carla Mendes       3687
Fábio Ramos        3674
Inês Nunes         3655
Diego Ferreira       11
Ana Lima              8
 Elisa Costa          7
 João Vieira          6
 Fábio Ramos          6
 Inês Nunes           6
Inês Nunes            6
 Carla Mendes         5
 Ana Lima             5
 Bruno Souza          5
Carla Mendes          4
Bruno Souza           4
João Vieira           4
Gabi Alves            4
Elisa Costa           4
 Gabi Alves           4
 Diego Ferreira       4
Henrique Dias         4
Fábio Ramos           3
 Henrique Dias        2
Name: count, dtype: int64

In [112]:
df['vendedor'] = df['vendedor'].fillna('Desconhecido')

Como haviam apenas 40 valores faltantes na coluna "vendedor", preencher com "Desconhecido" é de bom tom e não afeta nenhuma análise que seja feita em cima desses dados. A príncipio pensei em atribuir as vendas faltantes ao vendedor que mais fez vendas se fossem valores muitos distoantes (o 1º vendedor tendo feito 3000 vendas e o restante ter feito menos de 100), mas desisti ao perceber ao perceber que os top 10 haviam um número muito parecido de vendas.

##### C. Criação de novas colunas

In [120]:
df['custo_total']= custo_total
df['valor_bruto_venda'] = valor_bruto_venda
df['valor_desconto'] = valor_desconto
df['lucro_estimado'] = lucro_estimado
df['markup'] = markup
df['faixa_de_lucro']=df['lucro_estimado'].apply(lambda x: 'baixo' if x < 500 else 'médio' if x < 1000 else 'alto')
df

,data_venda,mes,cidade,canal_venda,categoria,produto,vendedor,fidelidade_cliente,preco_unitario,custo_unitario,margem_pct,quantidade,desconto_pct,receita_total,frete,avaliacao_cliente,prazo_entrega_dias,devolucao,entrega_no_prazo,alta_temporada,flag_outlier_preco,custo_total,valor_bruto_venda,valor_desconto,lucro_estimado,markup,faixa_de_lucro
0,2022-04-13,4,Rio de Janeiro,App Mobile,Eletrônicos,Notebook,Bruno Souza,diamante,739.39,561.75,22.31,10,5.2,6933.48,8.68,4.92,5.0,False,True,False,False,5617.50,7393.90,384.48280,1315.98,1.316226,alto
1,2023-03-12,3,Manaus,Site Web,Vestuário,Camiseta,Elisa Costa,bronze,187.48,135.15,30.26,5,0.6,930.28,NaN,4.98,19.0,True,False,False,False,675.75,937.40,5.62440,254.53,1.387199,baixo
2,2022-09-28,9,Goiânia,App Mobile,Vestuário,Tênis,João Vieira,bronze,76.11,61.40,21.61,14,0.7,1058.50,7.98,3.47,NaN,True,True,False,False,859.60,1065.54,7.45878,198.90,1.239577,baixo
3,2022-04-17,4,Fortaleza,Site Web,Vestuário,Calça Jeans,Carla Mendes,bronze,145.86,88.53,41.27,11,0.6,1592.43,20.49,4.50,16.0,False,True,False,False,973.83,1604.46,9.62676,618.60,1.647577,médio
4,2022-03-13,3,Belo Horizonte,App Mobile,Esportes,Bicicleta,João Vieira,bronze,133.92,107.25,9.47,1,19.5,107.64,22.89,3.49,13.0,False,True,False,False,107.25,133.92,26.11440,0.39,1.248671,baixo
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
37612,2023-12-31,3,Porto Alegre,App Mobile,Alimentação,Barra de Cereal,Gabi Alves,bronze,73.11,47.38,26.52,9,20.3,524.49,8.15,4.95,NaN,False,False,False,False,426.42,657.99,133.57197,98.07,1.543056,baixo
37613,2023-12-31,9,Salvador,Site Web,Alimentação,Mel Orgânico,Henrique Dias,bronze,76.62,55.50,22.31,12,10.7,820.52,41.74,4.49,19.0,False,False,False,False,666.00,919.44,98.38008,154.52,1.380541,baixo
37614,2023-12-31,2,Salvador,Loja Física,Alimentação,Whey Protein,Fábio Ramos,bronze,55.15,36.94,29.95,7,0.0,384.44,85.35,4.02,14.0,False,True,False,False,258.58,386.05,0.00000,125.86,1.492962,baixo
37615,2023-12-31,11,São Paulo,Site Web,Esportes,Tapete de Yoga,João Vieira,bronze,197.09,167.23,10.91,7,10.9,1960.18,30.99,3.50,6.0,False,True,True,False,1170.61,1379.63,150.37967,789.57,1.178556,médio


##### E. Agrupamentos com groupby

In [124]:
df.columns

Index(['data_venda', 'mes', 'cidade', 'canal_venda', 'categoria', 'produto',
       'vendedor', 'fidelidade_cliente', 'preco_unitario', 'custo_unitario',
       'margem_pct', 'quantidade', 'desconto_pct', 'receita_total', 'frete',
       'avaliacao_cliente', 'prazo_entrega_dias', 'devolucao',
       'entrega_no_prazo', 'alta_temporada', 'flag_outlier_preco',
       'custo_total', 'valor_bruto_venda', 'valor_desconto', 'lucro_estimado',
       'markup', 'faixa_de_lucro'],
      dtype='str')

In [142]:
df.groupby('categoria')['receita_total'].mean().sort_values(ascending=False)

categoria
Eletrônicos    5990.349951
Esportes       1512.880035
Vestuário       894.757710
Livros          602.565208
Alimentação     445.878146
Name: receita_total, dtype: float64

In [150]:
média= df.groupby('categoria')['receita_total'].mean()
print(f'a categoria de maior receita foi: {média.idxmax()}, cuja receita média foi de: {round(média.max(), 2)}')

a categoria de maior receita foi: Eletrônicos, cuja receita média foi de: 5990.35


In [152]:
df.groupby('cidade')['receita_total'].mean()

cidade
 Belo Horizonte    1310.652857
 Curitiba          3543.383333
 Fortaleza         1964.337143
 Goiânia            929.013333
 Manaus            2350.002857
 Porto Alegre      1164.280000
 Recife            1564.630000
 Rio de Janeiro    3338.010000
 Salvador          1006.915000
 São Paulo         1833.500000
B. Horizonte       2858.794444
BH                  750.220833
Belo Horizonte     1880.017159
Belo Horizonte     1705.802000
Belo horizonte     1587.480833
Curitiba           1821.459080
Curitiba           2025.202500
Curtiba            1550.238500
Fortaleza          1880.308225
Fortaleza          1046.815000
Goiânia            1892.116511
Goiânia            5068.018571
Manaus             1962.428366
Manaus              883.766250
Porto Alegre       1891.989111
Porto Alegre       3208.675000
RJ                  454.820000
Recife             2018.608007
Recife             2575.336000
Rio                1022.700909
Rio De Janeiro      838.178333
Rio de Janeiro     1853.894735
R

In [158]:
df.groupby('categoria')['lucro_estimado'].mean().sort_values(ascending=0).round(2)

categoria
Eletrônicos    1622.39
Esportes        409.78
Vestuário       244.06
Livros          165.49
Alimentação     120.98
Name: lucro_estimado, dtype: float64

In [159]:
df.groupby('faixa_de_lucro')['lucro_estimado'].mean().sort_values(ascending=0).round(2)

faixa_de_lucro
alto     2547.44
médio     704.91
baixo     138.15
Name: lucro_estimado, dtype: float64

##### F. Merge

In [184]:
df.cidade.unique()

<StringArray>
['rio de janeiro',         'manaus',        'goiânia',      'fortaleza',
 'belo horizonte',         'recife',      'são paulo',   'porto alegre',
       'salvador',       'curitiba']
Length: 10, dtype: str

In [189]:
regionalizacaoinator= pd.DataFrame({'cidade': ['são paulo', 'rio de janeiro', 'belo horizonte', 'curitiba',
               'porto alegre', 'salvador', 'recife', 'fortaleza','goiânia',
               'brasília', 'manaus'],'regiao': ['sudeste']*3 + ['sul']*2 + ['nordeste']*3 + ['centro-oeste']*2 + ['norte']})
regionalizacaoinator

,cidade,regiao
0,são paulo,sudeste
1,rio de janeiro,sudeste
2,belo horizonte,sudeste
3,curitiba,sul
4,porto alegre,sul
5,salvador,nordeste
6,recife,nordeste
7,fortaleza,nordeste
8,goiânia,centro-oeste
9,brasília,centro-oeste


In [191]:
df = df.merge(regionalizacaoinator, on='cidade', how='left')
df

,data_venda,mes,cidade,canal_venda,categoria,produto,vendedor,fidelidade_cliente,preco_unitario,custo_unitario,margem_pct,quantidade,desconto_pct,receita_total,frete,avaliacao_cliente,prazo_entrega_dias,devolucao,entrega_no_prazo,alta_temporada,flag_outlier_preco,custo_total,valor_bruto_venda,valor_desconto,lucro_estimado,markup,faixa_de_lucro,regiao
0,2022-04-13,4,rio de janeiro,App Mobile,Eletrônicos,Notebook,Bruno Souza,diamante,739.39,561.75,22.31,10,5.2,6933.48,8.68,4.92,5.0,False,True,False,False,5617.50,7393.90,384.48280,1315.98,1.316226,alto,sudeste
1,2023-03-12,3,manaus,Site Web,Vestuário,Camiseta,Elisa Costa,bronze,187.48,135.15,30.26,5,0.6,930.28,NaN,4.98,19.0,True,False,False,False,675.75,937.40,5.62440,254.53,1.387199,baixo,norte
2,2022-09-28,9,goiânia,App Mobile,Vestuário,Tênis,João Vieira,bronze,76.11,61.40,21.61,14,0.7,1058.50,7.98,3.47,NaN,True,True,False,False,859.60,1065.54,7.45878,198.90,1.239577,baixo,centro-oeste
3,2022-04-17,4,fortaleza,Site Web,Vestuário,Calça Jeans,Carla Mendes,bronze,145.86,88.53,41.27,11,0.6,1592.43,20.49,4.50,16.0,False,True,False,False,973.83,1604.46,9.62676,618.60,1.647577,médio,nordeste
4,2022-03-13,3,belo horizonte,App Mobile,Esportes,Bicicleta,João Vieira,bronze,133.92,107.25,9.47,1,19.5,107.64,22.89,3.49,13.0,False,True,False,False,107.25,133.92,26.11440,0.39,1.248671,baixo,sudeste
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
37612,2023-12-31,3,porto alegre,App Mobile,Alimentação,Barra de Cereal,Gabi Alves,bronze,73.11,47.38,26.52,9,20.3,524.49,8.15,4.95,NaN,False,False,False,False,426.42,657.99,133.57197,98.07,1.543056,baixo,sul
37613,2023-12-31,9,salvador,Site Web,Alimentação,Mel Orgânico,Henrique Dias,bronze,76.62,55.50,22.31,12,10.7,820.52,41.74,4.49,19.0,False,False,False,False,666.00,919.44,98.38008,154.52,1.380541,baixo,nordeste
37614,2023-12-31,2,salvador,Loja Física,Alimentação,Whey Protein,Fábio Ramos,bronze,55.15,36.94,29.95,7,0.0,384.44,85.35,4.02,14.0,False,True,False,False,258.58,386.05,0.00000,125.86,1.492962,baixo,nordeste
37615,2023-12-31,11,são paulo,Site Web,Esportes,Tapete de Yoga,João Vieira,bronze,197.09,167.23,10.91,7,10.9,1960.18,30.99,3.50,6.0,False,True,True,False,1170.61,1379.63,150.37967,789.57,1.178556,médio,sudeste


In [193]:
df.groupby('regiao')['lucro_estimado'].mean().round(2)

regiao
centro-oeste    515.97
nordeste        524.93
norte           543.85
sudeste         508.24
sul             485.99
Name: lucro_estimado, dtype: float64

In [195]:
df.groupby('regiao')['quantidade'].mean().round(2)

regiao
centro-oeste    7.38
nordeste        7.56
norte           7.42
sudeste         7.45
sul             7.50
Name: quantidade, dtype: float64

In [196]:
df.groupby('regiao')['receita_total'].max()

regiao
centro-oeste    51003.49
nordeste        48490.44
norte           34629.54
sudeste         58430.28
sul             22670.13
Name: receita_total, dtype: float64

## Entregável 3 Seaborn